### PR TITLE
feat: Add strict trace continuation support

### LINKED ISSUE #1005: Implement strict trace continuation (org_id validation)
## Summary

Implement the `strictTraceContinuation` option as specified in the [SDK spec](https://develop.sentry.dev/sdk/foundations/trace-propagation/#strict-trace-continuation).

This prevents unintended trace mixing across different Sentry organizations by validating the `org_id` in incoming baggage headers.

### 1. Parse org_id from DSN
Extract the organization ID from the DSN host (e.g., `o1234` from `o1234.ingest.sentry.io` → `"1234"`).

### 2. Add `org_id` configuration option
Allow explicit override of the DSN-derived org_id (needed for self-hosted / Relay setups).

### 3. Propagate org_id in baggage
Add `sentry-org_id` to the Dynamic Sampling Context (DSC) / baggage when creating head-of-trace.

### 4. Implement trace continuation validation
Add a `strict_trace_continuation` option (default: `false`) that validates incoming `sentry-org_id` against the SDK's org_id before continuing a trace. See the [spec](https://develop.sentry.dev/sdk/foundations/trace-propagation/#strict-trace-continuation) for the full decision matrix.

### PR DESCRIPTION
**Pull request by giortzisg**
## Summary
- Extract org ID from DSN host (e.g., `o1234.ingest.sentry.io` → `"1234"`) and propagate as `sentry-org_id` in outgoing baggage headers
- Validate incoming traces against the SDK's org ID to prevent cross-organization trace mixing
- Add `:org_id` config option for explicit override (self-hosted/Relay setups)
- Add `:strict_trace_continuation` config option (default `false`) — when enabled, both org IDs must be present and match

## Decision Matrix
| Baggage org | SDK org | strict=false | strict=true |
|-------------|---------|-------------|-------------|
| 1           | 1       | Continue    | Continue    |
| None        | 1       | Continue    | New trace   |
| 1           | None    | Continue    | New trace   |
| None        | None    | Continue    | Continue    |
| 1           | 2       | New trace   | New trace   |

## Changes
- `lib/sentry/dsn.ex` — org ID extraction from host using `^o(\d+)\.` regex
- `lib/sentry/config.ex` — `:org_id`, `:strict_trace_continuation` options + `effective_org_id/0`
- `lib/sentry/opentelemetry/propagator.ex` — `sentry-org_id` in baggage inject/extract + `should_continue_trace?/1`
- `test/sentry/strict_trace_continuation_test.exs` — DSN extraction, config, and full decision matrix tests (12 cases)

## Test plan
- [x] DSN org ID extraction tests (5 cases)
- [x] Config option default and override tests (4 cases)
- [x] effective_org_id precedence tests (4 cases)
- [x] Decision matrix tests (10 cases + 2 edge cases)
- [x] CI validation

Closes #1005

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/sentry/opentelemetry/propagator.ex [hunk 1]
```diff
@@ -56,19 +57,27 @@ if Sentry.OpenTelemetry.VersionChecker.tracing_compatible?() do
...
-              remote_span_ctx = :otel_tracer.from_remote_span(trace_id, span_id, 1)
+                # Create a remote, sampled parent span in the OTEL context.
+                # We will set to "always sample" because Sentry will decide real sampling
+                remote_span_ctx = :otel_tracer.from_remote_span(trace_id, span_id, 1)
 
-              Tracer.set_current_span(ctx, remote_span_ctx)
+                Tracer.set_current_span(ctx, remote_span_ctx)
+              else
+                require Logger
+                Logger.debug("[Sentry] Not continuing trace due to org ID mismatch")
```
[the remark this answers is not in the document]
Comment (solnic) [reply lib/sentry/opentelemetry/propagator.ex]: Probably including the other org id could be useful?

**Review by solnic — DISMISSED** [review]
This also requires a high level integration test.

File: lib/sentry/opentelemetry/propagator.ex [hunk 2]
```diff
@@ -131,5 +146,65 @@ if Sentry.OpenTelemetry.VersionChecker.tracing_compatible?() do
       missing = total_bytes - byte_size(bin)
       if missing > 0, do: :binary.copy(<<0>>, missing) <> bin, else: bin
     end
+
+    # Ensure sentry-org_id is present in the baggage string
+    defp ensure_org_id_in_baggage(baggage) when is_binary(baggage) do
+      org_id = Sentry.Config.effective_org_id()
+
+      if org_id != nil and extract_baggage_org_id(baggage) == nil do
+        baggage <> ",sentry-org_id=" <> org_id
```
[the remark this answers is not in the document]
Comment (solnic) [reply lib/sentry/opentelemetry/propagator.ex]: Good catch sentry bot 🙇🏻

**Review by solnic — APPROVED** [review]

**Review by dingsdax — APPROVED** [review]
